In [1]:
from transformers import GPT2LMHeadModel

In [2]:
model_hf = GPT2LMHeadModel.from_pretrained('gpt2')  # 124M
sd_hf = model_hf.state_dict()  # all the raw tensors

for k, v in sd_hf.items():
    # k -> different params of the gpt2 models
    print(k, v.shape)

config.json:   0%|          | 0.00/665 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/548M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

transformer.wte.weight torch.Size([50257, 768])
transformer.wpe.weight torch.Size([1024, 768])
transformer.h.0.ln_1.weight torch.Size([768])
transformer.h.0.ln_1.bias torch.Size([768])
transformer.h.0.attn.c_attn.weight torch.Size([768, 2304])
transformer.h.0.attn.c_attn.bias torch.Size([2304])
transformer.h.0.attn.c_proj.weight torch.Size([768, 768])
transformer.h.0.attn.c_proj.bias torch.Size([768])
transformer.h.0.ln_2.weight torch.Size([768])
transformer.h.0.ln_2.bias torch.Size([768])
transformer.h.0.mlp.c_fc.weight torch.Size([768, 3072])
transformer.h.0.mlp.c_fc.bias torch.Size([3072])
transformer.h.0.mlp.c_proj.weight torch.Size([3072, 768])
transformer.h.0.mlp.c_proj.bias torch.Size([768])
transformer.h.1.ln_1.weight torch.Size([768])
transformer.h.1.ln_1.bias torch.Size([768])
transformer.h.1.attn.c_attn.weight torch.Size([768, 2304])
transformer.h.1.attn.c_attn.bias torch.Size([2304])
transformer.h.1.attn.c_proj.weight torch.Size([768, 768])
transformer.h.1.attn.c_proj.bias 

`transformer.wte.weight torch.Size([50257, 768])`: 

- the token embeddings
- with 50257 tokens in gpt2 vocabularies, 
- and 768 dimensional embeddings, the vector represents each token

`transformer.wpe.weight torch.Size([1024, 768])`:

- lookup tables for positions, the `max_seq_length` of gpt 2 is 1024
- each token can be attending to up to 1024 positions
- each of the positions in gpt2 has a fixed vector of 768, learned by optimisation

In [12]:
from transformers import pipeline, set_seed
generator = pipeline('text-generation', model='gpt2')
set_seed(5525)
generator('Mayday is a rock band from China,', max_length=55, num_return_sequences=5)

Hardware accelerator e.g. GPU is available in the environment, but no `device` argument is passed to the `Pipeline` object. Model will be on CPU.
Truncation was not explicitly activated but `max_length` is provided a specific value, please use `truncation=True` to explicitly truncate examples to max length. Defaulting to 'longest_first' truncation strategy. If you encode pairs of sequences (GLUE-style) with the tokenizer you can select this strategy more precisely by providing a specific strategy to `truncation`.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.


[{'generated_text': 'Mayday is a rock band from China, releasing their first solo album after finishing their second album in 2013.\n\nThe album, which is also an online-only concert album, starts with this song : "Songs are what make it."\n\n"Just like'},
 {'generated_text': 'Mayday is a rock band from China, and they played at the annual Rock and Roll Hall of Fame. On Nov. 29, 2009, they broke out of an 18-month hiatus following a breakup with their manager, Guo Wen, and took center stage with "'},
 {'generated_text': 'Mayday is a rock band from China, who recently performed with a band from China once. They came up with the name in honor of the planet\'s most visited and popular planet. They have a record called "Ting Ting" (Sushi), and they are'},
 {'generated_text': 'Mayday is a rock band from China, with its roots in the West, that is known for their rock music. It has two founding members, Sugeun and Zhi-Song. Both of them have been members of the band for a long time. Suge'},
 